Change to InputEventMIDI event to include connection_index property

ramdor opened the pull request:
Added connection_index to the InputEventMIDI so that each MIDI message can be associated with a specific opened MIDI device.

- *Production edit: This closes https://github.com/godotengine/godot-proposals/issues/7733.*

ramdor committed:
Change to InputEventMIDI event to include connection_index property

Added connection_index to the InputEventMIDI so that each MIDI message can be associated with a specific opened MIDI device.

ramdor committed:
Fix typo in updated InputEventMIDI xml

typo fix

Calinou commented:
Does this close https://github.com/godotengine/godot-proposals/issues/7733?

ramdor commented:
@Calinou yes. The connection index is direcly related to the count through the midi devices that are opened.

ramdor committed:
clang-format update

clang-format applied

ramdor committed:
fix for int to void*

Quick fix for the compile error which I wasnt able to test locally easily. Hopefully this will now pass.

AThousandShips commented on core/input/input_event.h:
```diff
@@ -547,6 +548,9 @@ class InputEventMIDI : public InputEvent {
 	void set_controller_value(const int p_controller_value);
 	int get_controller_value() const;
 
+	void set_connection_index(const int p_connection_index);
```
```suggestion
	void set_connection_index(int p_connection_index);
```
While it's used elsewhere here we don't use `const` for simple types so let's not use it for new methods

AThousandShips commented on core/input/input_event.cpp:
```diff
@@ -1717,6 +1717,14 @@ int InputEventMIDI::get_controller_value() const {
 	return controller_value;
 }
 
+void InputEventMIDI::set_connection_index(const int p_connection_index) {
```
```suggestion
void InputEventMIDI::set_connection_index(int p_connection_index) {
```

akien-mga commented:
Shouldn't this be stored in the existing `InputEvent.device` property instead of adding a new parameter everywhere?

ramdor commented:
I can make all the changes and swap it to use InputEvent.device np, removing the need for the extra parameter and fix the documentation.

ramdor committed:
snake_case + using device from InputEvent

Uses InputEvent.device instead, removing the need for a new property in InputEventMIDI. Updated to snake_case.

ramdor commented:
thanks for the review, and for the advice on using the InputEvent device instead. I have made the tweaks and submitted the changes. Cheers.
